src/azure-cli/azure/cli/command_modules/role/_multi_api_adaptor.py

```diff
@@ -47,8 +47,12 @@ def create_role_assignment(self, client, assignment_name, role_id, object_id, sc
             'RoleAssignmentProperties' if self.old_api else 'RoleAssignmentCreateParameters',
             mod='models', operation_group='role_assignments')
         parameters = RoleAssignmentCreateParameters(role_definition_id=role_id, principal_id=object_id)
-        if assignee_principal_type:
-            parameters.principal_type = assignee_principal_type
+
+        # In 2022-04-01 API, principal_type is by default 'User', so we have to explicitly set it to None if we can't
+        # resolve principal type from Graph
+        # https://github.com/Azure/azure-rest-api-specs/issues/21664
```

Comment:
This is under discussion in issue https://github.com/Azure/azure-rest-api-specs/issues/21664 and email: _Microsoft.Authorization 2022-04-01 principalType has default value User_